lib/ansible/modules/windows/win_service.ps1

```diff
@@ -19,87 +19,438 @@
+        Fail-Json $result "Can only set desktop_interact to true when service is run with or 'username' equals 'LocalSystem'"
+    }
+
+    if ($start_mode -ne $null) {
+        Set-ServiceStartMode -svc $svc -start_mode $start_mode
+    }
+
+    if ($username -ne $null) {
+        Set-ServiceAccount -wmi_svc $wmi_svc -username $username -password $password
+    }
+
+    if ($display_name -ne $null) {
+        Set-ServiceDisplayName -svc $svc -display_name $display_name
+    }
+
+    if ($desktop_interact -ne $null) {
+        Set-ServiceDesktopInteract -wmi_svc $wmi_svc -desktop_interact $desktop_interact
+    }
+
+    if ($description -ne $null) {
+        Set-ServiceDescription -svc $svc -description $description
+    }
+
+    if ($path -ne $null) {
+        Set-ServicePath -name $svc.Name -path $path
+    }
+
+    if ($dependencies -ne $null) {
+        Set-ServiceDependencies -wmi_svc $wmi_svc -dependency_action $dependency_action -dependencies $dependencies
+    }
+
+    if ($state -ne $null) {
+        Set-ServiceState -svc $svc -wmi_svc $wmi_svc -state $state
+    }
+}
+
+$svc = Get-Service -Name $name -ErrorAction SilentlyContinue
+if ($svc) {
+    Set-ServiceConfiguration -svc $svc
+} else {
+    $result.exists = $false
+    if ($state -ne 'absent') {
+        # Check if path is defined, if so create the service
+        if ($path -ne $null) {
+            try {
+                if ($check_mode) {
+                    New-Service -Name $name -BinaryPathname $path -WhatIf
+                } else {
+                    New-Service -Name $name -BinaryPathname $path
+                }
+            } catch {
+                Fail-Json $result $_.Exception.Message
+            }
+            $result.changed = $true
+
+            $svc = Get-Service -Name $name
+            Set-ServiceConfiguration -svc $svc
+        } else {
+            # We will only reach here if the service is installed and the state is not absent
+            # Will check if any of the default actions are set and fail as we cannot action it
+            if ($start_mode -ne $null -or
+                $state -ne $null -or
+                $username -ne $null -or
+                $password -ne $null -or
+                $display_name -ne $null -or
+                $description -ne $null -or
+                $desktop_interact -ne $false -or
+                $dependencies -ne $null -or
+                $dependency_action -ne 'set') {
+                    Fail-Json $result "Service '$name' is not installed, need to set 'path' to create a new service"
+            }
+        }
+    }
+}
+
+# After making a change, let's get the service info again unless we deleted it
+if ($state -eq 'absent') {
+    # Recreate result so it doesn't have the extra meta data now that is has been deleted
+    $changed = $result.changed
+    $warnings = $result.warnings
+    $result = @{
+        changed = $changed
+        warnings = $warnings
+        exists = $false
```

Comment:
I'm not sure if there is an easier way, all it is doing is recreating an empty result again, do you have another suggestion?